watcher_namespaces.go

```diff
@@ -280,8 +299,10 @@ func (nw *NamespaceWatcher) refresh(infoKeys []string, rawMetrics map[string]str
 	for _, ns := range infoKeys {
 		nsName := strings.ReplaceAll(ns, "namespace/", "")
 
-		namespaceObserver := make(MetricMap, len(namespaceRawMetrics))
-		for m, t := range namespaceRawMetrics {
+		namespaceMetrics := getWhitelistedMetrics(namespaceRawMetrics, config.AeroProm.NamespaceMetricsWhitelist, config.AeroProm.NamespaceMetricsWhitelistEnabled)
```

Comment:
One thing that I forgot yesterday is that you are filtering the metrics on every refresh. This is inefficient and will get even slower when we introduce regex to the whitelists.
I think it's a better idea to introduce a new list like `filteredMetrics` on `init()` and filter the metrics once there, and just use the filtered one on the next iterations.